### PR TITLE
✨Allow custom text in ASAA CTA

### PR DESCRIPTION
**Pull request by calebcordry**
Previously we only allowed CTA text from predefined choices, this allows ads to have custom text.

The new behavior is that the button will grow horizontally to fit the new variable text length. If the text is too long to fit within the design spec, the ad will be discarded.

### CODE REVIEW

**Comment by calebcordry** [comment]
@lannka @powerivq friendly ping :) cc/ @katrinivini

File: extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    this.resources_ = Services.resourcesForDoc(ampdoc);
+
+    /** @private {!Document} */
+    this.doc_ = ampdoc.win.document;
+
+    /** @private {!Element} */
+    this.measurer_ = this.doc_.createElement('div');
+
+    /** @private {number} Fixed button height from design spec. */
+    this.maxHeight_ = 32;
```
Comment (powerivq) [inline extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js]: Is this always fixed? Should it defined as a constant?

File: extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ */
+function calculateFontSize(
+  measurer,
+  expectedHeight,
+  expectedWidth,
+  minFontSize,
+  maxFontSize
+) {
+  maxFontSize++;
+  // Binomial search for the best font size.
```
Comment (powerivq) [inline extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js]: I think it should be 
s/Binomial/Binary/

File: extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+/**
+ * Stolen from amp-fit-text.js
+ * @param {Element} measurer
+ * @param {number} expectedHeight
+ * @param {number} expectedWidth
+ * @param {number} minFontSize
+ * @param {number} maxFontSize
+ * @return {number}
+ */
+function calculateFontSize(
```
Comment (powerivq) [inline extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js]: Is this for future proofing? The current range is very small at 12-14

File: extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js (hunk 2 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js]: Removed with the fit logic changes.

File: extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js (hunk 1 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js]: Moved to const

File: extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js (hunk 3 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js]: Good idea, refactored to start at 14 and work down

Comment (powerivq) [reply extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js]: A bit detailed, but I think a for loop would be more readable.

Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/story-ad-button-text-fitter.js]: changed to for loop.

**Pull request merged**